File: data/code_pairs/fortran/2239.f90
PROGRAM Main
  CALL subr1(3.0, 4.0)
  CALL subr2
END PROGRAM Main

INTEGER FUNCTION ifun(arg1, arg2)
  INTEGER, INTENT(IN) :: arg1, arg2
  ifun = arg1 + arg2
END FUNCTION ifun

DOUBLE PRECISION FUNCTION jfun(arg1)
  DOUBLE PRECISION, INTENT(IN) :: arg1
  jfun = SQRT(arg1)
END FUNCTION jfun

SUBROUTINE subr1(a, b)
  DOUBLE PRECISION, INTENT(IN) :: a, b
  DOUBLE PRECISION :: c
  DOUBLE PRECISION :: arrayA(3,4,3)
  c = MAX(a, b)
  arrayA(1,1,1) = c
  PRINT *, 'Value is ', c
END SUBROUTINE subr1

SUBROUTINE subr2
  DOUBLE PRECISION :: i, l
  INTEGER :: j, m
  PRINT *, 'inside subroutine subr2'
  i = 1.0
  j = 2
  m = 3
  l = 4.5
  PRINT *, 'IFUN RETURNS ', ifun(j, m)
  PRINT *, 'JFUN RETURNS ', jfun(i + l)
END SUBROUTINE subr2

File: data/code_pairs/cpp/2239.cpp
#include <iostream>
#include <cmath>
#include <algorithm>

int ifun(int arg1, int arg2) {
    return arg1 + arg2;
}

double jfun(double arg1) {
    return std::sqrt(arg1);
}

void subr1(double a, double b) {
    double arrayA[3][4][3];
    double c = std::max(a, b);
    arrayA[0][0][0] = c;
    std::cout << "subr1 - Value is: " << c << std::endl;
}

void subr2() {
    double i = 1.0;
    int j = 2;
    int m = 3;
    double l = 4.5;

    std::cout << "inside subroutine subr2" << std::endl;
    std::cout << "IFUN RETURNS: " << ifun(j, m) << std::endl;
    std::cout << "JFUN RETURNS: " << jfun(i + l) << std::endl;
}

int main() {
    subr1(3.0, 4.0);
    subr2();
    return 0;
}